miner/worker.go

```diff
@@ -396,19 +392,10 @@ func (w *worker) mainLoop() {
 	defer wg.Wait()
 	txsCh := make(chan core.NewTxsEvent, txChanSize)
 
-	generateNewBlock := func(nextBlockNum *big.Int) {
+	generateNewBlock := func() {
 		if cancel != nil {
 			cancel()
 		}
-
-		if w.chainConfig.IsL2Migration(nextBlockNum) {
-			if w.isRunning() {
-				log.Info("The next block is the L2 migration block, stopping block construction", "currentBlock", w.chain.CurrentBlock().NumberU64(), "hash", w.chain.CurrentBlock().Hash(), "nextBlock", nextBlockNum.Uint64())
-				w.stop()
-			}
-			return
-		}
```

Comment:
With this removed, could the local node produce blocks after the migration? Or are those also checked by the block fetcher header validity check?

We also wouldn't stop the worker anymore. Is that a problem (resource usage, irritating logs or rpc responses)?

Reply:
I don't think the local node can produce blocks any more, because if you look at consensus we can see that `sendPreprepareV2` is the point at which the proposer sends the initial preprepare message.

```
func (c *core) sendPreprepareV2(request *istanbul.Request, roundChangeCertificateV2 istanbul.RoundChangeCertificateV2) {
	logger := c.newLogger("func", "sendPreprepareV2")

	// If I'm the proposer and I have the same sequence with the proposal
	if c.current.Sequence().Cmp(request.Proposal.Number()) == 0 && c.isProposer() {
		m := istanbul.NewPreprepareV2Message(&istanbul.PreprepareV2{
			View:                     c.current.View(),
			Proposal:                 request.Proposal,
			RoundChangeCertificateV2: roundChangeCertificateV2,
		}, c.address)
		logger.Debug("Sending preprepareV2", "m", m)
		c.broadcast(m)
	}
}

```

If you dig down into the `broadcast`  function you can see that the preprepare message eventually gets sent back to the proposer, for it to handle in the same way as all the other messages eventually arriving at `handlePreprepareV2` which calls `verifyProposal` on the block.


So to summarise consensus nodes, including proposers would not be able to progress past the preprepare phase of consensus if the block does not pass validation, and therefore not be able to produce new blocks.


>  We also wouldn't stop the worker anymore. Is that a problem (resource usage, irritating logs or rpc responses)?

I think that's a benefit, because we don't end up messing about with the internal stopping mechanisms.

Reply:
Sounds good, thanks for the explanation!